src/plugins/intel_gpu/src/plugin/ops/search_sorted.cpp

```diff
@@ -0,0 +1,23 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/op/search_sorted.hpp"
+
+#include "intel_gpu/plugin/program_builder.hpp"
+#include "intel_gpu/primitives/search_sorted.hpp"
+
+namespace ov {
+namespace intel_gpu {
+
+static void CreateSearchSortedOp(ProgramBuilder& p, const std::shared_ptr<ov::op::v15::SearchSorted>& op) {
+    validate_inputs_count(op, {2});
+    auto inputs = p.GetInputInfo(op);
+    auto roi_align_prim = cldnn::search_sorted(layer_type_name_ID(op), inputs[0], inputs[1], op->get_right_mode());
+    p.add_primitive(*op, roi_align_prim);
```

Comment:
As you are using `impl_param.desc->output_data_types[0].value_or(data_types::i64);` for shape inference, output_data_types have to be specified during primitive creation:
```suggestion
    p.add_primitive(*op, roi_align_prim);
    p.output_data_types = get_output_data_types(op);
```

Reply:
Fixed, but if it "has to be specified", why did it work before?(unittests, funct test, even on some model)???

Reply:
With the previous approach it wouldn't "not work", it will just ignore the data type specified for this operation in ov::model because the default data_type was always used, which may cause additional reorders in some cases